File: gaia-sdk-javascript/src/lib/atlas/__tests__/AtlasRequest.test.ts
import {AtlasQueryRequest, AtlasRequest} from '../AtlasRequest';

describe('AtlasRequestTest', () => {

    test('builds ner custom queries', () => {
        const request = AtlasRequest.query((it: AtlasQueryRequest) => {
            it.nlu("text", false, (nlu) => {
                nlu.ner((it: any) => {
                    it.custom('some-qualifier', (it: any) => {
                        it.data();
                        it.indices();
                    });
                });
            });
        });

        const [statement, variables] = request.getStatement();
        expect(statement).toEqual('query atlas($text1:String!, $merge1:Boolean!, $qualifier1:String!) { nlu(text:$text1, merge:$merge1) { ner { custom(qualifier:$qualifier1) { data indices } } } }');
        expect(variables).toEqual({"merge1":false, "qualifier1":"some-qualifier", "text1":"text"});
        expect(request.preprocessors).toEqual([])
    });

    test('builds multiple ner custom queries', () => {
        const request = AtlasRequest.query((it: AtlasQueryRequest) => {
            it.nlu("text", false, (nlu) => {
                nlu.ner((it: any) => {
                    it.custom('qualifier-1', (it: any) => {
                        it.data();
                        it.indices();
                    });
                    it.custom('qualifier-2', (it: any) => {
                        it.data();
                    });
                    it.custom('qualifier-3', (it: any) => {
                        it.indices();
                        it.negation();
                    });
                });
            });
        });

        const [statement, variables] = request.getStatement();
        expect(statement).toEqual('query atlas($text1:String!, $merge1:Boolean!, $qualifier1:String!, $qualifier2:String!, $qualifier3:String!) { nlu(text:$text1, merge:$merge1) { ner { custom(qualifier:$qualifier1) { data indices } custom(qualifier:$qualifier2) { data } custom(qualifier:$qualifier3) { indices negation } } } }');
        expect(variables).toEqual({"merge1":false, "qualifier1":"qualifier-1", "qualifier2":"qualifier-2", "qualifier3":"qualifier-3", "text1":"text"});
    });
});

File: gaia-sdk-python/atlas/test/AtlasRequestTest.py
import unittest

from atlas.AtlasRequest import QueryLex, QueryDep, QueryNlu, AtlasRequest, AtlasQueryRequest


class AtlasRequestTest(unittest.TestCase):

    def test_instantiation(self):
        def lex(x: QueryLex):
            x.lemma()
            x.base()
            x.pos()
            x.abbreviations(lambda x: x.lemma())
            x.flexions(lambda x: x.lemma())
            x.hyperonyms(lambda x: x.lemma())
            x.meronyms(lambda x: x.lemma())
            x.parts(lambda x: x.lemma())
            x.causations(lambda x: x.lemma())
            x.collocations(lambda x: x.lemma())
            x.synonyms(lambda x: x.lemma())
            x.labels()

        def dep(x: QueryDep):
            x.source()
            x.target()
            x.sourceBase()
            x.targetBase()
            x.sourceIndex()
            x.targetIndex()
            x.sourcePos()
            x.targetPos()
            x.sourceTag()
            x.targetTag()
            x.relation()
            x.rawRelation()

        def nlu(e: QueryNlu):
            e.raw()
            e.txt()
            e.cls("default")
            e.lex(lex)
            e.dep(dep)

        def config(query: AtlasQueryRequest):
            query.nlu("text", False, nlu)

        request = AtlasRequest.query(config)
        (statement, variables) = request.getStatement()

        self.assertEqual(statement,
                         "query atlas($text1:str, $merge1:bool, $qualifier1:str) {nlu(text:$text1, merge:$merge1) { raw txt cls(qualifier:$qualifier1) lex { lemma base pos abbreviations { lex { lemma } } flexions { lex { lemma } } hyperonyms { lex { lemma } } meronyms { lex { lemma } } parts { lex { lemma } } causations { lex { lemma } } collocations { lex { lemma } } synonyms { lex { lemma } } labels } dep { source target sourceBase targetBase sourceIndex targetIndex sourcePos targetPos sourceTag targetTag relation rawRelation } }}")


if __name__ == '__main__':
    unittest.main()
